src/Components/Articles/SingleArticle.jsx

```diff
@@ -0,0 +1,56 @@
+import { useState, useEffect } from "react";
+import { useParams } from "react-router-dom";
+import dateFormat from "dateformat";
+import { fetchArticleById } from "../../api";
+import Loader from "../Loader";
+
+function SingleArticle() {
+  const [article, setArticle] = useState({});
+  const [isLoading, setIsLoading] = useState(true);
+  const [isError, setIsError] = useState(false);
+  const { article_id } = useParams();
+
+  const formatDate = dateFormat(article.created_at, "dddd, dd mmmm yyyy 'at' H:MM");
+
+  const checkSingleVote = article.votes === 1 ? "like" : "likes";
+
+  useEffect(() => {
+    setIsLoading(true);
+    setIsError(false);
+
+    fetchArticleById(article_id)
+      .then((articleData) => {
+        console.log(articleData);
+        setArticle(articleData);
+        setTimeout(() => setIsLoading(false), 500);
+      })
+      .catch((err) => {
+        setIsError(true);
+        console.log(err);
+      });
+  }, [article_id]);
+
+  if (isError) {
+    return <h2>Something went wrong!</h2>;
+  }
+
+  if (isLoading) {
+    return <Loader />;
+  }
+
+  return (
+    <div className="article-container prose">
+      <h1>{article.title}</h1>
+      <img src={article.article_img_url} alt="" />
```

Comment:
I am not sure how to dynamically add alt text for each article image, is there a quick way of doing this? I think this is part of the advanced tasks so can look at this later. Will it involve adding an alt text row to the database, so that when future article images are added the alt text can be called from the api?